In [15]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
file_to_load = "../Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [16]:
#PLAYER COUNT
#count total players using SN column and print into dataframe
number_players = purchase_data["SN"].nunique()
Total_players= pd.DataFrame({"Total_players": [number_players] })
Total_players

Total_players
0            576

In [17]:
#PURCHASING ANALYSIS
#Obtain number of unique items, average price, etc using nunique,count,sum, and mean functions and display in dataframe.
Item_ID = purchase_data["Item ID"].nunique()
Item_ID_Total= purchase_data["Item ID"].count()
Total_Revenue= purchase_data["Price"].sum()
Average_Price = purchase_data["Price"].mean()

Purchasing_analysis= pd.DataFrame({"Unique Items": [Item_ID],"Item ID Total ": [Item_ID_Total], 
                                   "Total Revenue": [Total_Revenue], "Average Price": [Average_Price]})
#Format so it looks pretty<3
Purchasing_analysis["Average Price"]=Purchasing_analysis["Average Price"].astype(float).map(
"${:,.2f}".format)
Purchasing_analysis["Total Revenue"]=Purchasing_analysis["Total Revenue"].astype(float).map(
"${:,.2f}".format)


Purchasing_analysis



Unique Items  Item ID Total  Total Revenue Average Price
0           179             780     $2,379.77         $3.05

In [18]:
# Create Gender Demographics Percentage
Gender_percentage= purchase_data['Gender'].value_counts(normalize=True) * 100
#Create Gender Demographics Count
Gender_count= purchase_data['Gender'].value_counts()
Gender_Demo= pd.DataFrame({"Gender Percentage":Gender_percentage, "Gender count":Gender_count})

#Format Gender% to two decimal places
Gender_Demo["Gender Percentage"]=Gender_Demo["Gender Percentage"].astype(float).map(
"%{:,.2f}".format)

Gender_Demo

Gender Percentage  Gender count
Male                             %83.59           652
Female                           %14.49           113
Other / Non-Disclosed             %1.92            15

In [19]:
#PURCHASING ANALYSIS (Gender)
#describe(by gender) ;) using groupby and count/mean functions and then display results in DF

Gender_Group= purchase_data.groupby("Gender")
Purchase_count= Gender_Group["Purchase ID"].count()
Purchase_Price_Average= Gender_Group["Price"].mean()
Purchase_Total= Gender_Group["Item ID"].count()
Purchasing_analysis_gender= pd.DataFrame({"Purchase Count": Purchase_count,
"Average Purchase Price": Purchase_Price_Average, "Purchase Total":Purchase_Total
                                         })
#Format so it looks pretty
#Convert the avg purchase price column to float and adjust values to 2 decimal places with $ sign. 
Purchasing_analysis_gender["Average Purchase Price"]=Purchasing_analysis_gender["Average Purchase Price"].astype(float).map(
"${:,.2f}".format)


Purchasing_analysis_gender


Purchase Count Average Purchase Price  Purchase Total
Gender                                                                      
Female                            113                  $3.20             113
Male                              652                  $3.02             652
Other / Non-Disclosed              15                  $3.35              15

In [20]:
#AGE DEMOGRAPHICS

#Before creating bins for ages first find the youngest, middle, and oldest ages.
Ages= purchase_data["Age"].describe() 
Ages

#Establish bins for ages
Age_Demographics= pd.DataFrame(purchase_data)
Age_range = [0, 12.9,19.9, 24.9, 39.9, 45 ]

#Catogerize players by "generation" using age bins
Group_generations= ["Kids", "Teenagers", "Gen Z", "Millennials","Gen X"]
Age_Demographics["Age Demographics"]= pd.cut(Age_Demographics["Age"], Age_range,labels= Group_generations, include_lowest= True)



Age_Demographics= Age_Demographics.groupby("Age Demographics")

Total= Age_Demographics["SN"].nunique().sum()

Total_count= Age_Demographics["SN"].count()
#ungrouped data for deno 
Age_demographic_percentage = (Age_Demographics["Age Demographics"].value_counts()/Total)*100

Age_demographic_percentage = Age_demographic_percentage.droplevel(0)




Purchasing_age_demographic= pd.DataFrame({"Total count": Total_count,

                                          "Age Demographic Percentage":Age_demographic_percentage   
                                         })
# Format clean up
#Convert the avg age demographic percentage column to float and adjust values to 2 decimal places with $ sign. 
Purchasing_age_demographic[ "Age Demographic Percentage"]=Purchasing_age_demographic[ "Age Demographic Percentage"].astype(float).map(
"${:,.2f}".format)

Purchasing_age_demographic

Total count Age Demographic Percentage
Age Demographics                                        
Kids                       45                      $7.81
Teenagers                 142                     $24.65
Gen Z                     365                     $63.37
Millennials               215                     $37.33
Gen X                      13                      $2.26

In [21]:
#PURCHASING ANALYSIS BY AGE
#Before creating bins for ages first find the youngest, middle, and oldest ages.
Ages= purchase_data["Age"].describe() 
Ages

#Establish bins for ages
Age_Demographics= pd.DataFrame(purchase_data)
Age_range = [0, 12.9,19.9, 24.9, 39.9, 45 ]

#Catogerize players by "generation" using age bins
Group_generations= ["Kids", "Teenagers", "Gen Z", "Millennials","Gen X"]
By_Age_Group= Age_Demographics["Age Demographics"]= pd.cut(Age_Demographics["Age"], Age_range,labels= Group_generations, include_lowest= True)


# Group calculations by age demographic
Age_Group= Age_Demographics.groupby("Age Demographics")

Purchase_count_Age= Age_Group["Purchase ID"].count()
Purchase_Price_Average_Age= Age_Group["Price"].mean()
Purchase_Total_Age= Age_Group["Item ID"].count()
Average_Total_Per_Person= purchase_data["Price"].mean()

Purchasing_analysis_age= pd.DataFrame({"Purchase Count": Purchase_count_Age,
"Avg Purchase Price": Purchase_Price_Average_Age, "Purchase Total":Purchase_Total_Age, "Average Purchase Per Person":Average_Total_Per_Person
                                          
                                         })
#Cleaner Formating  
Purchasing_analysis_age["Avg Purchase Price"]= Purchasing_analysis_age["Avg Purchase Price"].astype(float).map(
"${:,.2f}".format)
Purchasing_analysis_age["Average Purchase Per Person"]= Purchasing_analysis_age["Average Purchase Per Person"].astype(float).map(
"${:,.2f}".format)
Purchasing_analysis_age


Purchase Count Avg Purchase Price  Purchase Total  \
Age Demographics                                                      
Kids                          45              $3.19              45   
Teenagers                    142              $3.02             142   
Gen Z                        365              $3.05             365   
Millennials                  215              $3.04             215   
Gen X                         13              $2.94              13   

                 Average Purchase Per Person  
Age Demographics                              
Kids                                   $3.05  
Teenagers                              $3.05  
Gen Z                                  $3.05  
Millennials                            $3.05  
Gen X                                  $3.05

In [22]:
#TOP SPENDERS
#Group data by SN

Spenders_SN= purchase_data.groupby("SN")
#Assign variables and run basic calculations
Purchase_price= Spenders_SN["Item ID"].nunique()
Item_Price= Spenders_SN["Price"].mean()
Total_value= Purchase_price*Item_Price
#Create dataframe and  sort total purchase value from highest to lowest price.
Top_Spenders= pd.DataFrame({"Purchase Count":Purchase_price, "Item Price":Item_Price, "Total Purchase Value":Total_value})
Spenders_df= Top_Spenders.sort_values("Total Purchase Value", ascending= False)

#Make Dataframe look pretty 
Spenders_df["Total Purchase Value"]= Spenders_df["Total Purchase Value"].astype(float).map("${:,.2f}".format)
Spenders_df["Item Price"]= Spenders_df["Item Price"].astype(float).map("${:,.2f}".format)

Spenders_df

Purchase Count Item Price Total Purchase Value
SN                                                         
Lisosia93                 5      $3.79               $18.96
Idastidru52               4      $3.86               $15.45
Chamjask73                3      $4.61               $13.83
Iral74                    4      $3.40               $13.62
Iskadarya95               3      $4.37               $13.10
...                     ...        ...                  ...
Ililsasya43               1      $1.02                $1.02
Irilis75                  1      $1.02                $1.02
Aidai61                   1      $1.01                $1.01
Chanirra79                1      $1.01                $1.01
Alo38                     1      $1.00                $1.00

[576 rows x 3 columns]

In [23]:
#MOST POPULAR ITEMS
#Locate columns 

Popular= purchase_data.set_index("Item ID", "Item Name","Price")
# Popular.head()

D_Name = purchase_data.groupby(["Item ID", "Item Name"])
P_count= D_Name["Item ID"].count()
Item_P=  D_Name["Price"].mean()
Total_Purchase_value= P_count*Item_P
Most_pop_items= pd.DataFrame({"Purchase Count":P_count, "Item Price":Item_P,"Total Purchase Value":Total_Purchase_value})
Items_df= Most_pop_items.sort_values("Purchase Count", ascending= False)

#Clean Data
Items_df["Total Purchase Value"]= Items_df["Total Purchase Value"].astype(float).map("${:,.2f}".format)
Items_df["Item Price"]= Items_df["Item Price"].astype(float).map("${:,.2f}".format)



Items_df

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   
...                                                              ...   
42      The Decapitator                                            1   
51      Endbringer                                                 1   
118     Ghost Reaver, Longsword of Magic                           1   
104     Gladiator's Glaive                                         1   
91      Celeste                                                    1   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
...                                                         ...   
42      The Decapitator                                   $1.75   
51      Endbringer                                        $4.66   
118     Ghost Reaver, Longsword of Magic                  $2.17   
104     Gladiator's Glaive                                $1.93   
91      Celeste                                           $4.17   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
...                                                                   ...  
42      The Decapitator                                             $1.75  
51      Endbringer                                                  $4.66  
118     Ghost Reaver, Longsword of Magic                            $2.17  
104     Gladiator's Glaive                                          $1.93  
91      Celeste                                                     $4.17  

[179 rows x 3 columns]

In [24]:
#MOST PROFITABLE ITEMS
#Locate columns 

Popular= purchase_data.set_index("Item ID", "Item Name","Price")
# Popular.head()

D_Name = purchase_data.groupby(["Item ID", "Item Name"])
P_count= D_Name["Item ID"].count()
Item_P=  D_Name["Price"].mean()
Total_Purchase_value= P_count*Item_P
Most_pop_items= pd.DataFrame({"Purchase Count":P_count, "Item Price":Item_P,"Total Purchase Value":Total_Purchase_value})
Items_df= Most_pop_items.sort_values("Total Purchase Value", ascending= False)

#Clean Data
Items_df["Total Purchase Value"]= Items_df["Total Purchase Value"].astype(float).map("${:,.2f}".format)
Items_df["Item Price"]= Items_df["Item Price"].astype(float).map("${:,.2f}".format)



Items_df

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   
...                                                              ...   
28      Flux, Destroyer of Due Diligence                           2   
125     Whistling Mithril Warblade                                 2   
126     Exiled Mithril Longsword                                   1   
104     Gladiator's Glaive                                         1   
42      The Decapitator                                            1   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   
...                                                         ...   
28      Flux, Destroyer of Due Diligence                  $1.06   
125     Whistling Mithril Warblade                        $1.00   
126     Exiled Mithril Longsword                          $2.00   
104     Gladiator's Glaive                                $1.93   
42      The Decapitator                                   $1.75   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80  
...                                                                   ...  
28      Flux, Destroyer of Due Diligence                            $2.12  
125     Whistling Mithril Warblade                                  $2.00  
126     Exiled Mithril Longsword                                    $2.00  
104     Gladiator's Glaive                                          $1.93  
42      The Decapitator                                             $1.75  

[179 rows x 3 columns]

Based on this data I will describe 3 observable trends:
1. the Final Critic is the most popular purchased Item. 
2. User Lisosia93purchased the most items at the price of $18.96.
3. This fantasy game is dominated by Males. 